### PR TITLE
Fix/save post after appending media item

### PR DESCRIPTION
**Pull request by mzorz**
This PR saves the post to FluxC right when a new media item is included in it, so to avoid more chances to lose the data while the media is uploaded but the Editor 

To test:
1. start a new draft
2. include some media item (or even a series of items you carefully selected from your phone)
3. stop the app
4. edit the post again
5. observe the media items are still there

cc @nbradbury

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -1811,12 +1813,21 @@ private boolean addMediaVisualEditor(Uri uri, boolean isVideo) {
...
 
         return true;
     }
 
+    private void appendMediaFileAndSavePost(MediaFile mediaFile, String imageUrl, ImageLoader imageLoader) {
+        mEditorFragment.appendMediaFile(mediaFile, imageUrl, mImageLoader);
+        // after asking the Editor to append the Media File, save the Post object to the DB
+        // as we need to keep the new modifications made to the Post's body (that is, the inserted media)
+        if (updatePostObject()) {
+            savePostToDb();
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Would it be better to call `savePostAsync()` here?

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java [hunk 2]
```diff
@@ -824,6 +824,9 @@ public void onMediaUploadSucceeded(final String localMediaId, final MediaFile me
                 AttributesWithClass attributesWithClass = new AttributesWithClass(
                         content.getElementAttributes(predicate));
                 attributesWithClass.removeClass(ATTR_STATUS_UPLOADING);
+                if (mediaFile.isVideo()) {
```
Comment (nbradbury) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java]: Just wanted to make sure you intended this change for this PR, since it seems unrelated.

Comment (mzorz) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java]: oh totally: it is. Just realized it was missing there only - should I create another PR for this change, if that makes sense?

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java (hunk 1 above)
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: tl;dr good call, addressed in 15fb128
I hesitated a bit on this as it seemed harmful that I only wanted to get the latest post data and  then disptach a FLuxC event (which is done asynchronously), but then realized the `updatePostObject`  could take long if this is a big post. Nice catch!

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java (hunk 2 above)
Comment (nbradbury) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java]: No need for a separate PR, just wanted to make sure it wasn't an accidental change!

**Comment by nbradbury** [comment]
:shipit:

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java (hunk 2 above)
Comment (mzorz) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java]: awesome, thanks!
